src/docfx/build/page/BuildPage.cs

```diff
@@ -375,5 +379,26 @@ private static string CreateContent(Context context, Document file, JObject page
             ValidateBookmarks(context, file, htmlDom);
             return CreateHtmlContent(context, htmlDom);
         }
+
+        private static void ValidateH1(Context context, Document file, string? title)
+        {
+            if (string.IsNullOrEmpty(title))
+                return;
+
+            var headings = new List<Heading>
+                {
+                    new Heading
+                    {
+                        Level = 1,
+                        Content = title,
+
+                        // todo: get title precise line info
+                        SourceInfo = new SourceInfo(file.FilePath, 0, 0),
+                    },
+                };
+
+            var validationContext = new ValidationContext { DocumentType = string.IsNullOrEmpty(file.Mime) ? "conceptual" : file.Mime.Value! };
+            context.DocsValidator.ValidateHeadings(headings, validationContext).GetAwaiter().GetResult();
```

Comment:
consider let validation library _not_ returning `Task`, I thought the design guideline for validation library is these methods are compute only, you are never suppose to do any network IO inside these methods. A non-task returning method signature helps convey that principle.

Reply:
Should `ValidateHeadings` return a validation results? If so, write the errors here.

Reply:
Just in case we need do some IO validation, like the current `Microsoft alias validation`.